### PR TITLE
Add 2 new Markdown IT Functionality (include and container)

### PR DESCRIPTION
**Pull request by L3houx**
- Add documentation in README (about the 2 functionalities that I added)
- Update package.json
- Modify src/System/Tasks/ConversionRunner.ts to add 2 Markdown-It package (include and container)
- Add a declare file to be able to compile the project du to the fact that markdown-it-include didn't had a @type

Thank You and Have a nice one!

### CODE REVIEW

File: src/System/Tasks/ConversionRunner.ts
Comment (manuth) [inline src/System/Tasks/ConversionRunner.ts:11]: Please remove this and use `upath`s `dirname` instead. It's already imported on line 15:
https://github.com/manuth/MarkdownConverter/blob/f3dde71d9455746939e230887f9ef05b60c9c1f0/src/System/Tasks/ConversionRunner.ts#L15

File: src/System/Tasks/ConversionRunner.ts
Comment (manuth) [inline src/System/Tasks/ConversionRunner.ts:523]: Please do not use the `var` keyword. Instead just pass the options directly:
```suggestion
        parser.use(
            markdownContainer,
            '',
            {
                validate: function (name: string) {
                    return name.trim().length;
                },
                render: function (token:Token[], id:number) {
                    if (token[id].info.trim() !== '') {
                        return `<div class="${token[id].info.trim()}">\n`;
                    } else {
                        return `</div>\n`;
                    }
                }
          });

        /**
         * Add Markdown IT Include
         */
        parser.use(
            markdownInclude,
            {
                root: path.dirname(window.activeTextEditor.document.fileName),
                includeRe: /\!{3}\s*include(.+?)\!{3}/i
            });
```

Comment (manuth) [inline src/System/Tasks/ConversionRunner.ts:523]: Please do not use the `var` keyword. Instead just pass the options directly:
```suggestion
        parser.use(
            markdownContainer,
            '',
            {
                validate: function (name: string) {
                    return name.trim().length;
                },
                render: function (token:Token[], id:number) {
                    if (token[id].info.trim() !== '') {
                        return `<div class="${token[id].info.trim()}">\n`;
                    } else {
                        return `</div>\n`;
                    }
                }
          });

        /**
         * Add Markdown IT Include
         */
        parser.use(
            markdownInclude,
            {
                root: path.dirname(window.activeTextEditor.document.fileName),
                includeRe: /\!{3}\s*include(.+?)\!{3}/i
            });
```

File: src/System/Tasks/ConversionRunner.ts
Comment (manuth) [inline src/System/Tasks/ConversionRunner.ts:525]: Please remove the duplicate new line character

File: package-lock.json
Comment (manuth) [inline package-lock.json:4]: Please install at least npm v7, completely revert all changes made to this file (`git checkout package-lock.json origin/dev`) and re-run `npm install`.
Having this `package-lock.json` file generated using `npm` with a version < 7 breaks the CI process (as npm v6 and lower **doesn't** install the `peerDependencies` of `dependencies` automatically while npm v7 and higher does).

I assume this is the reason for the CI process to fail.

**Review by manuth — CHANGES_REQUESTED** [review]
That's awesome! I've been looking for a convenient way to include other .md files and to apply CSS classes for long, tbh.

Before we can get started w/ merging this PR, please take note of the comments I've made and change your code appropriately.

In order to have your code checked correctly, you'll have to fix your code.

I assume the error that the CI is facing is caused by the `package-lock.json` file being generated using an `npm` version which is lower than v7.

Please perform following steps in order to fix it:
  - Revert changes made to `package-lock.json`: `git checkout origin/dev -- package-lock.json`
  - Install npm v7 (or higher - you might have to install a more recent version of nodejs in order to do so)
  - Re-run `npm install`

For a next time you might want to keep seperate features truly separate (i.e. create a separate PR for the CSS feature and a separate PR for the Include feature).

It's okay, though - we'll leave it as is for now.

Thanks a lot for the effort you put into this, I really appreciate it ✨

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: README.md
Comment (L3houx) [reply README.md]: Commit Resolved: **PR README Modifications** ( 9091c20bcc92ddd510cab427289098c1b3a4db51 )

File: src/System/Tasks/ConversionRunner.ts
Comment (L3houx) [reply src/System/Tasks/ConversionRunner.ts]: Commit Resolved: **PR Modifications ConversionRunner TS** ( 7d51275bb9492a485000963129e1df40a3f7b6ca )

File: src/System/Tasks/ConversionRunner.ts
Comment (L3houx) [reply src/System/Tasks/ConversionRunner.ts:523]: Commit Resolved: **PR Modifications ConversionRunner TS** ( 7d51275bb9492a485000963129e1df40a3f7b6ca )

File: package-lock.json
Comment (L3houx) [reply package-lock.json:4]: Commit Resolved: *PR Modifications Package Lock Json* ( d28ede37d4e09345c9f30da013b047c0021f09cc )

File: package.json
Comment (L3houx) [reply package.json:77]: Commit Resolved: *PR Modifications Package Lock Json* ( d28ede37d4e09345c9f30da013b047c0021f09cc )

**Comment by L3houx** [comment]
Thank a lot by the way to take the time to describe them all like this, it's really appreciated.  Once I did all you're recommandations, I update npm and nodejs to latests versions
node -v: `v17.2.0`
npm -v: `8.1.4`

I had this error, Locally:



Same as Build:
image




I was able to fix it using `hashFunction: "xxhash64"` in the webpack config. I will wait your feedback and if needed, I will push back the webpack config

**Comment by manuth** [comment]
@L3houx for some reason I don't know, the Drone CI build didn't start.
Anyways - there are still some linting issues. Could you please have a look at the output of `npm run lint`?

Thanks for taking care 😄

**Comment by L3houx** [comment]
`npm run lint`
image

I will check to fix these warnings

**Comment by L3houx** [comment]
Hi @manuth, I fixed the `npm run lint`

image

I will push my modifications soon.

**Comment by L3houx** [comment]
I have the same error as before, I'm waiting your call to update the webpack config to remove this error.

image

**Comment by manuth** [comment]
Oh I'm sorry - I totally forgot about that.
I'll gladly accept your changes made to the webpack config, then 😄

**Comment by manuth** [comment]
@L3houx running the tests failed.
I assume it's cause of this piece of code:
https://github.com/manuth/MarkdownConverter/blob/810fe455b56d1d4552130f62a529a0096899761a/src/System/Tasks/ConversionRunner.ts#L525

However - I wanted to change how detecting the root works anyways.
If you don't mind I'll push my proposed changes directly to your PR.

**Comment by L3houx** [comment]
It's all good for me, I will wait your commit to use it in the **markdown-it-include** extension configuration

**Comment by manuth** [comment]
@L3houx the CI tests will still fail
This is related to vscode applying IDs to headings by itself.

As this issue is unrelated to this very PR is ready to go.

Let me know if you are ready to have this PR merged or if you have something more to add.

**Comment by L3houx** [comment]
It's all good for me :) You can press the merge button :sunglasses:

**Comment by manuth** [comment]
Alright - did so! Thanks for your contribution, I really appreciate it 😄

**Pull request merged**
